specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,313 @@
+    if (dropHelper)
+    {
+        dropHelper->DragLeave();
+    }
+
+    return m_webViewCompositionController2->DragLeave();
+}
+
+HRESULT DropTarget::Drop(IDataObject* dataObject,
+    DWORD keyState,
+    POINTL point,
+    DWORD* effect)
+{
+    POINT point = { cursorPosition.x, cursorPosition.y };
+    // Tell the helper that we dropped onto it so it can update the drag image
+    // and get the correct effect.
+    wil::com_ptr<IDropTargetHelper> dropHelper = DropHelper();
+    if (dropHelper)
+    {
+        dropHelper->Drop(dataObject, &point, *effect);
+    }
+
+    // Convert the screen point to client coordinates add the WebView's offset.
+    // This returns whether the resultant point is over the WebView visual.
+    m_viewComponent->OffsetPointToWebView(&point);
+    return m_webViewCompositionController2->Drop(
+        dataObject, keyState, point, effect);
+}
+```
+
+## WinRT
+```c#
+// WinRT Sample
+private void WebView_DragEnter(object sender, DragEventArgs e)
+{
+  uint keyboardState = 
+    ConvertDragDropModifiersToWin32KeyboardState(e.Modifiers);
+  Point pointerPosition = CoreWindow.GetForCurrentThread().PointerPosition;
+  DataPackageOperation operation = 
+    (DataPackageOperation)webView2CompositionController.DragEnter(
+      e.Data, keyboardState, pointerPosition);
+  e.AcceptedOperation = operation;
+}
+
+private void WebView_DragOver(object sender, DragEventArgs e)
+{
+  uint keyboardState =
+    ConvertDragDropModifiersToWin32KeyboardState(e.Modifiers);
+  Point pointerPosition = CoreWindow.GetForCurrentThread().PointerPosition;
+  DataPackageOperation operation = 
+    (DataPackageOperation)webView2CompositionController.DragOver(
+      keyboardState, pointerPosition);
+  e.AcceptedOperation = operation;
+}
+
+private void WebView_DragLeave(object sender, DragEventArgs e)
+{
+  webView2CompositionController.DragLeave();
+}
+
+private void WebView_Drop(object sender, DragEventArgs e)
+{
+  uint keyboardState =
+    ConvertDragDropModifiersToWin32KeyboardState(e.Modifiers);
+  Point pointerPosition = CoreWindow.GetForCurrentThread().PointerPosition;
+  DataPackageOperation operation =
+    (DataPackageOperation)webView2CompositionController.Drop(
+      e.Data, keyboardState, pointerPosition);
+  e.AcceptedOperation = operation;
+}
+
+// Win32 keyboard state modifiers that are relevant during drag and drop.
+public const uint MK_LBUTTON = 1;
+public const uint MK_RBUTTON = 2;
+public const uint MK_SHIFT = 4;
+public const uint MK_CONTROL = 8;
+public const uint MK_MBUTTON = 16;
+public const uint MK_ALT = 32;
+
+// Helper function to convert DragDropModifiers to win32 keyboard state
+// modifiers that WebView2 uses during drag and drop operation.
+private uint ConvertDragDropModifiersToWin32KeyboardState(
+  Windows.ApplicationModel.DataTransfer.DragDrop.DragDropModifiers modifiers)
+{
+  uint win32DragDropModifiers = 0;
+  if ((modifiers & DragDropModifiers.Shift) == DragDropModifiers.Shift)
```

Comment:
More idiomatic C# would be

```suggestion
  if (modifiers.HasFlag(DragDropModifiers.Shift))
```

Reply:
Switched so we do conversion internally using winrt